### PR TITLE
Add 'CompatiblePSEditions' to PowerShell Core built-in modules

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Add `CompatiblePSEditions = @("Core")` to PowerShell Core built-in modules.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by rjmholt** [comment]
Do we control the PSReadLine psd1? I noticed trying it with my compatibility checking prototype that it also doesn't have this field 😱

File: src/Modules/Shared/Microsoft.PowerShell.Host/Microsoft.PowerShell.Host.psd1 [hunk 1]
```diff
@@ -4,10 +4,8 @@ Author="PowerShell"
 CompanyName="Microsoft Corporation"
 Copyright="Copyright (c) Microsoft Corporation. All rights reserved."
 ModuleVersion="3.0.0.0"
```
Comment (SteveL-MSFT) [inline src/Modules/Shared/Microsoft.PowerShell.Host/Microsoft.PowerShell.Host.psd1]: There's a request to update our in-package ModuleVersion https://github.com/PowerShell/PowerShell/issues/2136, I think we can just have the ModuleVersion match the PSCore version "6.1.0.0" for this release until we can actually decouple them.

**Comment by SteveL-MSFT** [comment]
@rjmholt That needs to be addressed in the PSReadLine repo.  However, since PSReadLine supports PSv3 and higher which doesn't understand the `CompatiblePSEditions` property, we can't add it.

**Comment by rjmholt** [comment]
Also, given that the modules have been changing in sync with PowerShell and that PowerShell is now version 6.1, should the `PowerShellVersion` field be `6.1`?

It's unlikely that someone would even manage to get to the point where loading old PowerShell modules into a newer version is possible, but it's conceivable, and in principle should be protected against maybe?

**Comment by rjmholt** [comment]
@SteveL-MSFT New PSReadLine will only support v5+ right? So that can have the field? But I'm assuming that until then, or in cases using older PSReadLine versions, we take advantage of implicit compatibility due to path location?

**Comment by SteveL-MSFT** [comment]
@rjmholt regarding `PowerShellVersion`, yes it would be more accurate to set it to `6.1`.  As for PSReadLine, I think even 2.0.0 still supports v3, not sure.  I would suggest following up on the PSReadLine repo.

**Comment by daxian-dbw** [comment]
@rjmholt @SteveL-MSFT Your comments were addressed, please take another look when you have time. Thanks!

**Comment by rjmholt** [comment]
@SteveL-MSFT Looks like PSReadLine 2.0 is now v5+: https://github.com/lzybkr/PSReadLine/blob/46960b1fe442024923381eedd1c9100ded30d123/PSReadLine/Changes.txt#L32

**Review by rjmholt — APPROVED** [review]

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**